src/build_system.ml

```diff
@@ -394,81 +400,84 @@ let set_rule_generators ~init ~gen_rules =
   Fdecl.set t.init_rules init_rules;
   Fdecl.set t.gen_rules gen_rules
 
-let get_dir_status t ~dir =
-  match Path.Table.find t.dirs dir with
-  | Some x -> Some x
+let get_dir_triage t ~dir =
+  match Path.as_in_source_tree dir with
+  | Some dir ->
+    Dir_triage.Known (Non_build (
+      Path.Source.Set.to_list (File_tree.files_of t.file_tree dir)
+      |> List.map ~f:Path.source
+      |> Path.Set.of_list))
```

Comment:
a function `Path.Set.of_source_paths : Source.Set.t -> t` would be useful. See string_of_paths for example.